Makefile

```diff
@@ -1,37 +1,79 @@
 # Copyright (c) 2024, The Pallene Developers
 # Pallene Tracer is licensed under the MIT license.
 # Please refer to the LICENSE and AUTHORS files for details
-# SPDX-License-Identifier: MIT 
+# SPDX-License-Identifier: MIT
 
-CC     := gcc
-CFLAGS := -DPT_DEBUG -O2 -std=c99 -pedantic -Wall -Wextra
+# Where to install our things
+PREFIX = /usr/local
+BINDIR = $(PREFIX)/bin
+INCDIR = $(PREFIX)/include
 
-LUA_DIR := /usr
+# Where to find Lua libraries
+LUA_PREFIX = /usr/local
+LUA_BINDIR = $(LUA_PREFIX)/bin
+LUA_INCDIR = $(LUA_PREFIX)/include
+LUA_LIBDIR = $(LUA_PREFIX)/lib
 
-INSTALL_DIR    := /usr/local
-INSTALL_INCDIR := $(INSTALL_DIR)/include
-INSTALL_LIBDIR := $(INSTALL_DIR)/lib
-INSTALL_BINDIR := $(INSTALL_DIR)/bin
+# How to install files
+INSTALL= install -p
+INSTALL_EXEC= $(INSTALL) -m 0755
+INSTALL_DATA= $(INSTALL) -m 0644
 
-.PHONY: install ptracer_header pt-run libptracer uninstall clean
+# C compilation flags
+CFLAGS   = -DPT_DEBUG -O2 -std=c99 -pedantic -Wall -Wextra
```

Comment:
I suppose we could add an extra `-Wformat-security` flag. Not important but I presume in GitHub Action environment the compiler use that flag and already is generating some warnings in the Actions job. This flag should warn us about using string buffers instead of string literals as format specifier in functions like `fprintf`.

Reply:
I've opened an issue. Let's do it in another PR. I don't want to mix it in the middle of larger changes.

Reply:
Nevermind. I had to fix this to get the CI to pass. (done!)